packages/next/src/server/app-render/types.ts

```diff
@@ -223,3 +223,8 @@ export type InitialRSCPayload = {
   /** GlobalError */
   G: React.ComponentType<any>
 }
+
+export type RSCPayload =
+  | InitialRSCPayload
+  | NavigationFlightResponse
+  | ActionFlightResponse
```

Comment:
Discussed this offline and decided this might be a good change to make in a future, separate PR in addition to improving the shape of `FlightDataPath`